internal/jobcontainers/oci.go

```diff
@@ -0,0 +1,77 @@
+package jobcontainers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/jobobject"
+
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/Microsoft/hcsshim/internal/oci"
+	"github.com/Microsoft/hcsshim/internal/processorinfo"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+)
+
+// This file contains helpers for converting parts of the oci spec to useful
+// structures/limits to be applied to a job object.
+
+// Oci spec to job object limit information. Will do any conversions to job object specific values from
+// their respective OCI representations. E.g. we convert CPU count into the correct job object cpu
+// rate value internally.
+func specToLimits(ctx context.Context, s *specs.Spec) (*jobobject.JobLimits, error) {
+	// CPU limits
+	cpuNumSet := 0
+	cpuCount := uint32(oci.ParseAnnotationsCPUCount(ctx, s, oci.AnnotationContainerProcessorCount, 0))
+	if cpuCount > 0 {
+		cpuNumSet++
+	}
+
+	cpuLimit := uint32(oci.ParseAnnotationsCPULimit(ctx, s, oci.AnnotationContainerProcessorLimit, 0))
+	if cpuLimit > 0 {
+		cpuNumSet++
+	}
+
+	cpuWeight := uint32(oci.ParseAnnotationsCPUWeight(ctx, s, oci.AnnotationContainerProcessorWeight, 0))
+	if cpuWeight > 0 {
+		cpuNumSet++
+	}
+
+	if cpuNumSet > 1 {
+		return nil, fmt.Errorf("invalid spec - Windows Job Container CPU Count: '%d', Limit: '%d', and Weight: '%d' are mutually exclusive", cpuCount, cpuLimit, cpuWeight)
+	} else if cpuNumSet == 1 {
+		hostCPUCount := uint32(processorinfo.ProcessorCount())
+		if cpuCount > hostCPUCount {
+			log.G(ctx).WithFields(logrus.Fields{
+				"requested": cpuCount,
+				"assigned":  hostCPUCount,
+			}).Warn("Changing user requested CPUCount to current number of processors")
+			cpuCount = hostCPUCount
+		}
```

Comment:
this could also be inside `cpuCount != 0`?

Reply:
Yep, would probably be better suited in there honestly